.github/workflows/ci.yml

```diff
@@ -0,0 +1,167 @@
+
+            - name: "Install PHPUnit"
+              run: vendor/bin/simple-phpunit install
+
+            - name: "PHPUnit version"
+              run: vendor/bin/simple-phpunit --version
+
+            - name: "Run tests"
+              run: vendor/bin/simple-phpunit
+
+    extension-tests:
+        needs:
+            - 'tests'
+
+        name: "${{ matrix.extension }} with PHP ${{ matrix.php-version }}"
+
+        runs-on: 'ubuntu-latest'
+
+        continue-on-error: true
+
+        strategy:
+            matrix:
+                php-version:
+                    - '7.2.5'
+                    - '7.3'
+                    - '7.4'
+                    - '8.0'
+                extension:
+                    - 'extra/cssinliner-extra'
+                    - 'extra/html-extra'
+                    - 'extra/inky-extra'
+                    - 'extra/intl-extra'
+                    - 'extra/markdown-extra'
+                    - 'extra/string-extra'
+
+        steps:
+            - name: "Checkout code"
+              uses: actions/checkout@v2.3.3
+
+            - name: "Install PHP with extensions"
+              uses: shivammathur/setup-php@2.7.0
+              with:
+                  coverage: "none"
+                  php-version: ${{ matrix.php-version }}
+                  ini-values: memory_limit=-1
+                  tools: composer:v2
+
+            - name: "Add PHPUnit matcher"
+              run: echo "::add-matcher::${{ runner.tool_cache }}/phpunit.json"
+
+            - name: "Set composer cache directory"
+              id: composer-cache
+              run: echo "::set-output name=dir::$(composer config cache-files-dir)"
+
+            - name: "Cache composer"
+              uses: actions/cache@v2.1.2
+              with:
+                  path: ${{ steps.composer-cache.outputs.dir }}
+                  key: ${{ runner.os }}-${{ matrix.php-version }}-${{ matrix.extension }}-${{ hashFiles('composer.json') }}
+                  restore-keys: ${{ runner.os }}-${{ matrix.php-version }}-${{ matrix.extension }}-
+
+            - run: composer install
+
+            - name: "Install PHPUnit"
+              run: vendor/bin/simple-phpunit install
+
+            - name: "PHPUnit version"
+              run: vendor/bin/simple-phpunit --version
+
+            - if: matrix.extension == 'extra/markdown-extra' && matrix.php-version == '8.0'
+              working-directory: ${{ matrix.extension}}
+              run: composer config platform.php 7.4.99
+
+            - name: "Composer install"
+              working-directory: ${{ matrix.extension}}
+              run: composer install
+
+            - name: "Run tests"
+              working-directory: ${{ matrix.extension}}
+              run: ../../vendor/bin/simple-phpunit
+
+    integration-tests:
+        needs:
+            - 'tests'
+
+        name: "Integration tests with PHP ${{ matrix.php-version }}"
+
+        runs-on: 'ubuntu-20.04'
+
+        continue-on-error: true
```

Comment:
`continue-on-error` means further steps will run even if the previous ones failed. This can lead to very weird errors in the report if the PHP setup fails for instance. I suggest removing it.
for cases where you have multiple test steps in the job (which is not the case here AFAIK), there is a better trick to ensure these steps all run: https://github.com/symfony/demo/pull/1171#discussion_r517236662

Reply:
You are right it is not needed here, but afaik this means to continue the workflow job, even if one job by the matrix fails.

https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#example-preventing-a-specific-failing-matrix-job-from-failing-a-workflow-run

It is not continuing the job itself if one step fails and that is what you mean right?

Reply:
> You are right it is not needed here, but afaik this means to continue the workflow job, even if one job by the matrix fails.

no. thats what `matrix.fail-fast: false` is about

Reply:
`continue-on-error` is about not making the job fail when a step fails

Reply:
> no. thats what matrix.fail-fast: false is about

I was aware of `env.fail-fast`, but not as an available variable of the `matrix`

I am confused, based on this: https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#example-preventing-a-specific-failing-matrix-job-from-failing-a-workflow-run

![CleanShot 2020-11-05 at 14 05 49](https://user-images.githubusercontent.com/995707/98244649-0ecc4b80-1f70-11eb-88a3-ac567ac08b72.png)

This is exactly what I want to do and do 🤔  `fail-fast` is to mark the whole job (including the whole matrix) mark as failed if one of them `fail`...

Reply:
> I was aware of `env.fail-fast`, but not as an available variable of the `matrix`

sorry, that's a mistake. It is a sibling of the matrix.

Reply:
no problem, but I don't want to fail fast. fail-fast can only be set (and makes only sense in this case) to `true` OR `false`, not based on a changing variable.

Lets say fail-fast: true` and I have 4 PHP versions to test -> 4 jobs.
If no the first job fails , job 2,3 & 4 gets never executed further and therefore skipped.
But I want to run all of them to see which of them are green and if it is maybe only failing due to a specific php version